### ISSUE TITLE
🐛 Bug: Display 'Add to calendar' email as the organizer doesn't work as exoected when I have the same calendar in two Google Accounts

### ISSUE THREAD

**Issue by andy-vdg** [calcom/cal.com#14052]
### Issue Summary

I have two Google accounts
andy@personal.com
andy@work.com

The andy@personal.com account has full access to the andy@work.com calendar. 
I want to check both calendars in cal.com but I also want to send the email as andy@work.com. 
So, I've attempted to add both accounts.

#### First Issue:
When I add my work account it takes me through the authentication process but never actually adds the account. I tried several times without success. I had to first turn off andy@work.com for the personal account and then it added it. 
![image](https://github.com/calcom/cal.com/assets/19990167/ad627dd9-5cdf-452f-93ba-681ee2361162)

#### Second Issue: 
Now that I have added both cal.com gets confused because it finds andy@work.com twice. Whenever I turn off checking for andy@work.com via my personal account it also turns it off on the work account. 
![image](https://github.com/calcom/cal.com/assets/19990167/b67bdd0d-40b7-4a8c-b0e8-fb4954fe91f9)

When I edit the event, it doesn't matter how many times I try to set it to add to the calendar andy@work.com via the **work** account it will always revert to andy@work.com via the **personal** account. Therefore, it always sends the emails from andy@personal.com. 

![image](https://github.com/calcom/cal.com/assets/19990167/838c4a86-9638-4437-abf9-edeaf8600a52)

### Actual Results

- Due to cal.com becoming confused by the duplicate calendar it doesn't send from the correct account. 

### Expected Results

- I would expect it to send from the work account when I select it from the dropdown and not revert to the personal one. 
- I am assuming this is because the calendars ID itself is the same in your backend and it just takes the first one it encounters. 

### Workaround

The current workaround that seems to work for me is to remove both accounts and then re-add the andy@work.com first and then add andy@personal.com second. 
![image](https://github.com/calcom/cal.com/assets/19990167/f3649404-d747-4f1e-be85-f3df87840546)
Went back into the Events and edited the Add to calendar dropdown and now it works. 
![image](https://github.com/calcom/cal.com/assets/19990167/b4b1d71f-d799-4853-bfc6-4da3a752b292)

**Comment by ciaranha** [comment]
@Udit-takkar i assume this issue arose solely as a result of needing to reconnect the accounts?

![image](https://github.com/calcom/cal.com/assets/4536123/737e63ea-adbd-4fb2-9ff7-c9e99e6660b9)

**Comment by andy-vdg** [comment]
Not exactly... this was happening **while** I was reconnecting the calendar accounts to use this feature.

**Comment by ciaranha** [comment]
> Not exactly... this was happening **while** I was reconnecting the calendar accounts to use this feature.

Thanks for reporting this @andy-vdg. @Udit-takkar can you take a look?
